tail/tail.go

```diff
@@ -191,12 +191,22 @@ func tailSingleFile(tailer *tail.Tail, file string, stateFile string) chan strin
 	}()
 
 	go func() {
-		for line := range tailer.Lines {
-			if line.Err != nil {
-				// skip errored lines
-				continue
+	ReadLines:
+		for {
+			select {
+			case line, ok := <-tailer.Lines:
+				if !ok {
+					// tailer.Lines is closed
+					break ReadLines
+				}
+				if line.Err != nil {
+					// skip errored lines
+					continue
+				}
+				lines <- strings.TrimSpace(line.Text)
+			case <-abort:
+				break ReadLines
```

Comment:
I believe select statements choose which channel from which to read randomly when both have something available.  I expect them to block when neither does, and when either channel gets a message to accept it immediately. If all that's right, then the order of the two case statements shouldn't matter.